bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/handler/HDPowerViewShadeHandler.java

```diff
@@ -84,6 +84,8 @@ private enum RefreshKind {
     private static final String DETECTED_TILT_ANYWHERE = "tiltAnywhereDetected";
     private static final ShadeCapabilitiesDatabase DB = new ShadeCapabilitiesDatabase();
 
+    private static final int HARDWIRED_POWER_SUPPLY = 1;
```

Comment:
This could be moved to `ShadeData` for encapsulation. Additionally, in that class a new method could be added that would either return an enum for the batteryKind (which would need to be introduced) or simply a boolean, e.g. `isHardwiredPowerSupply()`. This way data and interpretation of that data would be closely together.

Reply:
Yes I agree.

Reply:
I have implemented an enum